synapse/storage/event_federation.py

```diff
@@ -203,15 +203,36 @@ def get_rooms_with_many_extremities(self, min_count, limit):
         """
 
         def _get_rooms_with_many_extremities_txn(txn):
-            sql = """
```

Comment:
I would write this as

```
            sql = """
                SELECT room_id FROM event_forward_extremities
                WHERE %s
                GROUP BY room_id
                HAVING count(*) > ?
                ORDER BY count(*) DESC
                LIMIT ?
      """ % (where_clause,)
```
(having previously calculated the where_clause, obs. make it "1=1" for the base case.)

(I see you've got it in the HAVING clause at the moment. I'm not sure that is correct?)